text/0009-client-api.md

```diff
@@ -0,0 +1,154 @@
+# Title
+
+## Client API of Casper Node
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0009](https://github.com/casperlabs/ceps/pull/9)
+
+This describes the client-facing API of the Casper Node.  It will be used by all clients interacting
+with a Casper Node, including the Rust client, the Python client and Clarity.
+
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+It will comprise two main parts: an HTTP server using [JSON-RPC](https://www.jsonrpc.org/specification)s
+to satisfy client requests, and an HTTP server using
+[Server-Sent Events](https://en.wikipedia.org/wiki/Server-sent_events) (SSEs) to satisfy streaming
+events to subscribed clients.  The streamed events will be JSON-encoded data.
+
+A shift in paradigm from the Scala node is that the node will not retain state relating to deploys
+or blocks other than what is required to satisfy the correct operation of the node.  In other words,
+the node will not retain state purely to be able to satisfy client requests to retrieve such
+ancillary state.  Rather the node will present an event stream to which clients may subscribe,
+allowing them to build up the information they require.
+
+So, for example, rather than being able to return information like the cost of executing a deploy,
+or an error message resulting from failed execution, this information will be announced by the
+relevant component (the `BlockExecutor` in this case) and emitted via SSEs by the `SseServer`
+component.
```

Comment:
Does the `BlockExecutor` support query by `deployHash` or clients could only listen to the event stream?

There are some use cases that users want to know the cost or status of deploy that they deployed some days ago.

Reply:
Currently, we don't keep track of deploy status directly in node-rs. The comment I added above "...we may need to store some kind of executed deploy summary so that we can answer basic questions about deploy disposition, as a short term measure until such a time as we can offer a middleware implementation..." acknowledges that we might need to add some minimal state in the short to mid term until we have put more effort into the event stream concept.

Reply:
Thank you for your explanation. If `node-rs` could answer basic questions about deploy disposition, I think the client API is sufficient for Clarity.

Reply:
Great - thanks!